File: base/src/main/java/com/appoets/base/base/BaseBottomSheet.java
package com.appoets.base.base;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;

public abstract class BaseBottomSheet<T extends ViewDataBinding> extends BottomSheetDialogFragment {

    private View rootView;
    private T mViewDataBinding;
    private View view;

    protected abstract void initView(View View);

    public abstract int getLayout();

    public abstract void initView(ViewDataBinding mViewDataBinding);

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        mViewDataBinding = DataBindingUtil.inflate(inflater, getLayout(), container, false);
        initView(mViewDataBinding);
        view = mViewDataBinding.getRoot();
        return view;
    }
}

File: traximodule/src/main/java/com/appoets/gojek/traximodule/views/views/bottomsheets/BaseBottomSheet.kt
package com.appoets.gojek.traximodule.views.views.bottomsheets

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.android.material.bottomsheet.BottomSheetDialogFragment

abstract  class BaseBottomSheet : BottomSheetDialogFragment(){
    abstract fun getLayoutId(): Int
    abstract fun initView(view: View)
    private  var rootView:View?=null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (rootView == null)
        {
            rootView = inflater.inflate(getLayoutId(), container, false)
            initView(rootView!!)
        }

        return rootView
    }


}
